src/idl_gen_ts.cpp

```diff
@@ -0,0 +1,1616 @@
+    }
+
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      if (field.deprecated) continue;
+
+      const auto field_name = MakeCamel(field.name, false);
+      const std::string field_binded_method =
+          "this." + field_name + ".bind(this)";
+
+      std::string field_val;
+      std::string field_type;
+      // a string that declares a variable containing the
+      // offset for things that can't be generated inline
+      // empty otw
+      std::string field_offset_decl;
+      // a string that contains values for things that can be created inline or
+      // the variable name from field_offset_decl
+      std::string field_offset_val;
+      const auto field_default_val =
+          GenDefaultValue(field, "flatbuffers", imports);
+
+      // Emit a scalar field
+      const auto is_string = IsString(field.value.type);
+      if (IsScalar(field.value.type.base_type) || is_string) {
+        const auto has_null_default = is_string || HasNullDefault(field);
+
+        field_type += GenTypeName(imports, field, field.value.type, false,
+                                  has_null_default);
+        field_val = "this." + field_name + "()";
+
+        if (field.value.type.base_type != BASE_TYPE_STRING) {
+          field_offset_val = "this." + field_name;
+        } else {
+          field_offset_decl = GenNullCheckConditional(
+              "this." + field_name,
+              "builder.createString(this." + field_name + "!)", "0");
+        }
+      }
+
+      // Emit an object field
+      else {
+        auto is_vector = false;
+        switch (field.value.type.base_type) {
+          case BASE_TYPE_STRUCT: {
+            const auto &sd = *field.value.type.struct_def;
+            field_type += GetObjApiClassName(sd, parser.opts);
+
+            const std::string field_accessor = "this." + field_name + "()";
+            field_val = GenNullCheckConditional(field_accessor,
+                                                field_accessor + "!.unpack()");
+            field_offset_val = GenNullCheckConditional(
+                "this." + field_name, "this." + field_name + "!.pack(builder)",
+                "0");
+
+            break;
+          }
+
+          case BASE_TYPE_VECTOR: {
+            auto vectortype = field.value.type.VectorType();
+            auto vectortypename =
+                GenTypeName(imports, struct_def, vectortype, false);
+            is_vector = true;
+
+            field_type = "(";
+
+            switch (vectortype.base_type) {
+              case BASE_TYPE_STRUCT: {
+                const auto &sd = *field.value.type.struct_def;
+                field_type += GetObjApiClassName(sd, parser.opts);
+                field_type += ")[]";
+
+                field_val = GenBBAccess() + ".createObjList(" +
+                            field_binded_method + ", this." + field_name +
+                            "Length())";
+
```

Comment:
I don't understand the logic of a blank line insertion.
In this method, I see both dense code without blank lines and very sparse code with many blank lines.
Better to use blank lines only if it is needed.

Reply:
Agreed but again, that is preexisting code that I don't want to touch in this PR.

Reply:
Ok.
I reviewed this file as-is without comparison with deleted `idl_gen_js_ts.cpp`. So all my review notes might be unrelated to this PR.

Reply:
If there are no issues with the current PR, it would be great to merge it, since it's a blocker for 2.0.

Reply:
If merge it right now `git` will "drop" the history of deleted  'idl_gen_js_ts.cpp`. Git can't detect renaming of it into `idl_gen_ts.cpp` due to big changes.
One way is to make "renaining" commit before this (#6404).
Another - revet `idl_gen_ts.cpp` to 'idl_gen_js_ts.cpp`.

Reply:
No problem, @bjornharrtell can you open a PR with just file rename?

Reply:
@krojew that is what I have done with https://github.com/google/flatbuffers/pull/6404 :)

Reply:
btw I think this should be squash merged